packages/ui/src/views/repo/repo-commits/components/pull-request-commits.tsx

```diff
@@ -0,0 +1,78 @@
+import { useMemo } from 'react'
+
+import { Avatar, AvatarFallback, CommitCopyActions, NodeGroup, StackedList, Text } from '@/components'
+import { formatDate, getInitials } from '@/utils/utils'
+
+import { TypesCommit } from '../types'
+
+interface CommitProps {
+  data?: TypesCommit[]
+}
+
+export const PullRequestCommits = ({ ...props }: CommitProps) => {
+  const data = props.data
+  const commitsGroupedByDate: Record<string, TypesCommit[]> = useMemo(
+    () =>
+      data?.reduce(
+        (group, commit) => {
+          const date = formatDate(commit.committer?.when as string)
+          group[date] = (group[date] || []).concat(commit)
+          return group
+        },
+        {} as Record<string, TypesCommit[]>
+      ) || {},
+    [data]
+  )
+
+  const entries = Object.entries(commitsGroupedByDate)
+  const totalNodes = entries.length
+
+  return (
+    <>
+      {entries.map(([date, commitData], node_idx) => (
+        <NodeGroup.Root key={date}>
+          <NodeGroup.Icon simpleNodeIcon />
+          <NodeGroup.Title>{date && <Text color="tertiaryBackground">Commits on {date}</Text>}</NodeGroup.Title>
+          <NodeGroup.Content>
+            {commitData && commitData.length > 0 && (
```

Comment:
is it possible to have 0 commits at this point? if so, what does the UI look like in that scenario?